src/test/java/codeu/controller/AdminServletTest.java

```diff
@@ -63,27 +68,59 @@ public void testDoGet_NullUser() throws IOException, ServletException {
     Mockito.verify(mockResponse).sendRedirect("/login");
   }
 
+  /**
+   * Tests that when you get a user from the session and the user DOES NOT exist
+   * in our database, checks that user is redirected to /login.
+   */
+  @Test
+  public void testDoGet_FakeUser() throws IOException, ServletException {
+    Mockito.when(mockSession.getAttribute("user")).thenReturn("fake_username");
+    Mockito.when(mockUserStore.getUser("fake_username")).thenReturn(null);
+
+    adminServlet.doGet(mockRequest, mockResponse);
+
+    Mockito.verify(mockResponse).sendRedirect("/login");
+  }
+
 /*
   @Test
   public void testDoGet_ValidUser() throws IOException, ServletException {
+    User admin_user =
+        new User(
+            UUID.randomUUID(),
+            "admin_username",
+            "admin_password",
+            Instant.now());
+
     Mockito.when(mockSession.getAttribute("user")).thenReturn("admin_username");
+    Mockito.when(mockUserStore.getUser("admin_username")).thenReturn(admin_user);
 
     adminServlet.doGet(mockRequest, mockResponse);
 
-    Mockito.verify(mockSession).getAttribute("user").matches("admin_username");
     Mockito.verify(mockRequestDispatcher).forward(mockRequest, mockResponse);
```

Comment:
I got to the conclusion after reviewing every other test file, that when you want to redirect them somewhere else you use mockResponse, and when you only want forward the request you use mockRequestDispatcher, this method is commented though, because it was marking me an error.